src/clr-angular/utils/popover/popover-content.ts

```diff
@@ -68,6 +70,18 @@ export class ClrPopoverContent implements AfterContentChecked, OnDestroy {
         } else {
           this.removeContent();
         }
+      }),
+      this.smartPositionService.shouldRealign.subscribe(() => {
+        this.shouldRealign = true;
+        // Avoid flickering on initialization, caused by the asynchronous nature of the
+        // check-collector pattern.
+        this.renderer.setStyle(this.view.rootNodes[0], 'opacity', '0');
+      }),
+      // Here we collect subsequent synchronously received content-check events and only take action
+      // at the end of the cycle. See bellow for details on the check-collector pattern.
+      this.checkCollector.pipe(debounceTime(0)).subscribe(() => {
+        this.doAlign();
+        this.renderer.setStyle(this.view.rootNodes[0], 'opacity', '1');
```

Comment:
Should doAlign be responsible for setting it visible when its ready?

Reply:
It could, but:
- They are synchronous, so it's functionally the same.
- I preferred to keep "doAlign" focused on alignment stuff only; and this event handler as a controller method, that instructs: align, please; after this - return visibility.
- I also think it's better for maintenance, as it's close in code to the opacity: 0 change and it's easier to follow the relationship between the two.